filebeat/docs/how-does-filebeat-work.asciidoc

```diff
@@ -0,0 +1,76 @@
+
+=== What is an prospector?
+
+Filebeat currently supports two `prospector` types: `log` and `stdin`. Each prospector type can be defined multiple times. The responsibility of the prospector is to find all sources it should read from. In case of the log input type it is to find all files on the drive based on the defined paths and start a harvester for each file that was found. Each prospector looks as following:
+
+```
+filebeat.prospectors:
+- input_type: log
+  paths:
+    - /var/log/*.log
+    - /var/path2/*.log
+```
+
+The log prospector is responsible to find all files which were defined under the paths above and check, if for each file a harvester has to be started, if one is already running or the file can be ignored because of some configuration options. Each prospector runs in its own go routine and is responsible to manage the harvesters. New files are only picked up, if the offset / size of the file changed since the harvester was closed.
+
+=== What is a harvester?
+
+A harvester is responsible to read the content of a single file. That is done by reading each file line by line and send the information to the output. Each harvester is started by its prospector and retrieves all configuration options from the prospector. The harvester is responsible to open and close files. That means, as long as a harvester is running, the file descriptor stays open. Even if a file is removed or renamed, filebeat will keep reading the file. This has the side affect that the space on your disk will be reserved until the harvester is stopped. Stopping a harvester means:
+
+* The file handler is closed
+* The harvesting will only be picked up again after `scan_frequencey`
+* In case the file was moved / removed, reading the file will not continue
+
+
+=== What is the state?
+
+Filebeat keeps the state of each file and frequently flushs it to disk to the registry file. The state is used to remember the last offset a harvester was reading from and to ensure all log lines are sent. In case Elasticsearch or Logstash are down, Filebeat will keep the track of the last lines sent for each file and will continue reading the file as soon elasticsearch becomes available again. As long as filebeat is running, all this information is kept in memory and frequently written to disk to the `registry_file`. In case of a restart of filebeat, the data from the registry file is used to rebuild the state and filebeat continues at the last known position.
+
+Each prospector keeps a state for each file it finds. As files can be renamed or moved, the filename and path are not enough to identify a file. For each file unique identifiers are stored to detect if a file the same file that was harvested previously.
+
+
+== Use cases
+
+There are common use cases and problems which are described below on how they can be solved with the configuration options.
```

Comment:
The complete sentence is now not needed anymore, as I moved these paragraphs to troubleshooting, so the context is already given.